### DISCUSSION TITLE
Feature request - image display

### DISCUSSION THREAD

**Discussion by robmarkcole** [mu-editor/mu#1404] [Ideas & New Features]
I really love the live plotting mode, it is great for basic sensor data. However I also have a use case for plotting area date, e.g. from an array sensor such as https://learn.adafruit.com/adafruit-amg8833-8x8-thermal-camera-sensor/python-circuitpython
Can this be added as a feature request?

**Comment by ZanderBrown** [comment]
I'm not sure what your asking for?

Do you want different types of graph? If QChart supports it then I can't see any immediate reason why not

**Comment by robmarkcole** [comment]
Hi @ZanderBrown 
the data is a 32x32 numpy array being returned by an IR sensor. I would normallu plot it using matplotlib imshow [e.g](https://matplotlib.org/3.1.1/gallery/images_contours_and_fields/image_demo.html#sphx-glr-gallery-images-contours-and-fields-image-demo-py)

**Comment by aflaxman** [comment]
I'm interested in this feature request also, and I bet there is a way to do it with QtCharts.  Maybe QPixmap is the way to do it https://pythonbasics.org/pyqt-qpixmap/ or maybe there is a way to link matplotlib into the existing Qt panel https://matplotlib.org/faq/usage_faq.html#how-do-i-select-pyqt4-or-pyside

I can do some experimenting, but it would be great to get some hints.

**Comment by carlosperate** [comment]
A specialised plot might be outside of the Mu scope, but if the plot is generic enough and enough people request it we could consider its inclusion if somebody is happy to work on a PR.

In the meantime the Python mode REPL uses IPython, which includes matplotlib, and as Mu comes with PySerial as well it should be possible to create a simple script to get the serial data from devices and plot it with matplotlib.

**Comment by aflaxman** [comment]
Oh, nice.  Using matplotlib in the REPL is pretty close to what I want.  I see that if I use matplotlib with Run, it works also, and opens a new window:

```
import numpy as np, matplotlib.pyplot as plt
plt.imshow(np.random.uniform(size=(12,6)))
plt.show()
```

![image](https://user-images.githubusercontent.com/51236/92160049-44737d80-ede3-11ea-8f62-0a32da64c421.png)


This is probably just fine for my intended use-case, but I'll circle back to this issue if I need more.
